scalableminds/webknossos
# Add "merge" blend mode

### URL of deployed dev instance (used for testing):
- https://coverblendmode.webknossos.xyz/

#### TODOs
- [x] check all layer types
  - [x] uint24 doesn't work :/
    - [x] patched it, but need to talk with @philippotto about this "fix". (why the other packing degrees have a default alpah value of `10.0` and but the uint24 case does not. 
  - [x] Layer with `packingDegree = 2` / `byteCount = 2` is still untested. This is not included the data layers dataset.
- [x] build blend mode switch in frontend
  - [x] discuss name of this blend mode. GIMP calls it "merge" as it merges images together where layer 1 has one part of the image and the other layer has the rest of the image. But I think we agreed on calling this blend mode "cover" as the lower layers cover the layers above in the layer list.


### Steps to test:
- Open a dataset with multiple layers (e.g. the data layers dataset) in view mode
- Make all layers visible -> the data should be pretty bright
- Got to the settings tab to the data rendering settings section and change the render mode to cover.
- Now only the last layer in the list should be rendered.
- Turn all layers off besides the uint24 layer. This layer should also be rendered correctly.
- Then play a little around with what layers are set to visible and what rendering mode you select.
- Please notice, that both float layers are slightly darker than the other layers. This is likely due to the layers / data of the layers themselves (as this behaviour is already present on the master).

### Issues:
- fixes [#4194](https://github.com/scalableminds/webknossos/issues/4194)

------
(Please delete unneeded items, merge only when none are left open)
